### DISCUSSION TITLE
Disable Graphql

### DISCUSSION THREAD

**Discussion by nautiam** [requarks/wiki#5998] [Help / Questions] [answered]
Hi,

I don't want to public /graphql path. Is there anyway to disable GraphQL?

Thanks

**Comment by NGPixel** [comment, marked as answer]
No and it's not public. You need a valid token to access or perform any operation on it.

**Reply by nautiam** [reply]
I saw that it is public and I can access from public URL. For example, I can query all articles in wikijs and see their properties without token. Is there any way to forbid these types of operation?

**Reply by NGPixel** [reply]
That's incorrect. You can only query and see pages that are **public**. If you can query them without a token, it means they are accessible to the Guests group, whether it's via the GraphQL API or the website itself.

Show me a page that cannot be viewed directly but can be via the GraphQL?

**Reply by nautiam** [reply]
![ceph-adm](https://user-images.githubusercontent.com/5852412/209894277-0aff13db-4d1c-48c9-9428-d9e654433ed1.png)
![Screenshot from 2022-12-29 09-02-51](https://user-images.githubusercontent.com/5852412/209894280-152d3a4f-d044-4a0d-ab7d-e17afad0d8ab.png)
I attached the screenshot. As you can see, I open wikijs as anonymous in private mode and I can see the page ceph-adm without token, but I can't view or edit the content of page because I set it not publish.

**Reply by NGPixel** [reply]
If you list the `locale` property, you can see that the page you're referencing is in the `en` locale, not the `vi` as you posted.

public: https://docs.sunteco.vn/en/platform/ceph-adm
restricted: https://docs.sunteco.vn/vi/platform/ceph-adm

The one listed in your graphql result is the english one:
<img width="304" alt="image" src="https://user-images.githubusercontent.com/15522395/209895518-7679d47d-a07e-4f19-8fdc-236db3bc0b4d.png">

If you try to query the content of page ID 81, which is the one referring to the `vi` locale, you'll get a forbidden error:
<img width="950" alt="image" src="https://user-images.githubusercontent.com/15522395/209895803-fe0daad8-aa03-47ac-9286-25db7819a075.png">
